newrelic-api/src/main/java/com/newrelic/api/agent/NoOpAgent.java

```diff
@@ -168,6 +168,14 @@ public void acceptDistributedTracePayload(String payload) {
         @Override
         public void acceptDistributedTracePayload(DistributedTracePayload payload) {
         }
+
+        @Override
+        public void insertDistributedTraceHeaders(Map<String, String> headersMap) {
+        }
+
+        @Override
+        public void acceptDistributedTraceHeaders(HeaderType headerType, Map<String, String> headersMap) {
```

Comment:
We've discussed the accepting `TransportType` here, which is available in the agent bridge and would have to be promoted to be in the agent api module. I found the `HeaderType` enum already in the agent api module, and with a partially overlapping set of values as `TransportType`, and thought it might be more appropriate. 

Thoughts?

Reply:
I wonder why the TransportType has not been promoted already?  It offers more options to be specific and that would seem to be a good thing.  

I guess we could always add more constants to the HeaderType enum but that's potentially further repeating what's already on the TransportType.

Reply:
Another thing worth nothing is that `HeaderType` is already part of the contract of `InboundHeaders` and `OutboundHeaders`, which are used quite a bit throughout the API an the implementation.

Reply:
In this case would it make sense to enhance `HeaderType` (so that it has parity with `TransportType`) and just replace usages of `TransportType` with it?

Reply:
hmmm actually, I'm not sure if using `HeaderType` is the right choice as it deviates from the spec: https://source.datanerd.us/agents/agent-specs/blob/master/distributed_tracing/Distributed-Tracing.md#7-parent-transport-type

At a minimum `HeaderType` needs to have parity with all of the types declared in `TransportType`

Also this kind of convolutes matters given that all of our [metrics](https://source.datanerd.us/agents/agent-specs/blob/master/distributed_tracing/Distributed-Tracing.md#metrics) and [attributes](https://source.datanerd.us/agents/agent-specs/blob/master/distributed_tracing/Distributed-Tracing.md#attributes) explicitly refer to `transportType`.

I think promoting `TransportType` to the public API is probably the correct path forward.

Reply:
Yeah, `TransportType` seems like the right fit, but we may have to do some other not-so-minor refactors to accommodate it, possibly making breaking changes to the `InboundHeaders` and `OutboundHeaders` interfaces which are part of the public API.